helm/templates/deployment.yaml

```diff
@@ -19,42 +25,52 @@ spec:
       app: {{ $chartName }}
       release: {{ $releaseName }}
       run: {{ $releaseName }}-{{ $chartName }}
-      {{- include "ts-server-boilerplate.selectorLabels" . | nindent 6 }}
+      {{- include "overseer.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       labels:
         app: {{ $chartName }}
         release: {{ $releaseName }}
         run: {{ $releaseName }}-{{ $chartName }}
-        {{- include "ts-server-boilerplate.selectorLabels" . | nindent 8 }}
+        {{- include "overseer.selectorLabels" . | nindent 8 }}
       annotations:
-      {{- if .Values.resetOnConfigChange }}
+        {{- if .Values.resetOnConfigChange }}
         checksum/configmap: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
-      {{- end }}
-      {{- if .Values.podAnnotations }}
-      {{- toYaml .Values.podAnnotations | nindent 8 }}
-      {{- end }}
+        {{- end }}
+        {{- if $metrics.enabled }}
+        prometheus.io/port: {{ $metrics.prometheus.port | quote }}
+        prometheus.io/scrape: {{ $metrics.prometheus.scrape | quote }}
+        {{- end }}
+        {{- if .Values.podAnnotations }}
+        {{- toYaml .Values.podAnnotations | nindent 8 }}
+        {{- end }}
     spec:
-      {{- include "ts-server-boilerplate.imagePullSecrets" . | nindent 6 }}
+      {{- if $cloudProviderImagePullSecretName }}
+      imagePullSecrets:
+        - name: {{ $cloudProviderImagePullSecretName | quote }}
+      {{- end }}      
       containers:
         - name: {{ $releaseName }}-{{ $chartName }}-deployment
-          image: {{ include "ts-server-boilerplate.image" . }}
+          {{- with .Values.image }}
+          image: {{ $cloudProviderDockerRegistryUrl }}{{ .repository }}:{{ $imageTag }}
+          imagePullPolicy: {{ .pullPolicy | default "IfNotPresent" }}
+          {{- end }}
           {{- if .Values.command }}
           command: 
-          {{- toYaml .Values.command | nindent 12 }}
+            {{- toYaml .Values.command | nindent 12 }}
+          {{- end }}
           {{- if .Values.args }}
           args: 
-          {{- toYaml .Values.args | nindent 12 }}
-          {{- end }}
+            {{- toYaml .Values.args | nindent 12 }}
           {{- end }}
           volumeMounts:
             {{- if .Values.caSecretName }}
```

Comment:
change according to the new ca values (take is from global)

Reply:
done